src/query/api/v1/handler/prometheus/native/common_test.go

```diff
@@ -120,13 +146,28 @@ func TestParseDurationParsesIntAsSeconds(t *testing.T) {
 	assert.Equal(t, 30*time.Second, v)
 }
 
+func TestParseDurationParsesFloatAsSeconds(t *testing.T) {
```

Comment:
Could you add a case for `"/foo?step=30"`?

Reply:
There is one already.